builder/vsphere/common/step_keep_cdrom.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type KeepCDRomConfig
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
+)
+
+type KeepCDRomConfig struct {
+	// Keep a number of CD-ROM devices. Range: 1-4.
+	// If set to 0, `keep_cdrom` is ignored and the step is skipped.
+	// When set to a value in the range, `remove_cdrom` is ignored and
+	// the CD-ROM devices are kept without any attached media.
+	KeepCdrom int32 `mapstructure:"keep_cdrom"`
+}
+
+type StepKeepCDRom struct {
+	Config      *KeepCDRomConfig
+	CDRomConfig *CDRomConfig
+}
+
+func (s *StepKeepCDRom) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packersdk.Ui)
+	vm := state.Get("vm").(driver.VirtualMachine)
+
+	var err error
+	var errs []error
+
+	// Check if `keep_cdrom` is set.
+	keepCdrom := s.Config.KeepCdrom
+	if keepCdrom > 0 {
+		if keepCdrom < 1 || keepCdrom > 4 {
+			// If `keep_cdrom` is set, it must be between 1 and 4.
+			err := fmt.Errorf("'keep_cdrom' should be between 1 and 4")
+			errs = append(errs, err)
```

Comment:
We can actually get rid of this append because you return on line #48 so we are appending to the slice for nothing. 
```suggestion

```

Reply:
Resolved in f8dd848.